src/ray/core_worker/core_worker.cc

```diff
@@ -1153,8 +1153,11 @@ void CoreWorker::HandleKillActor(const rpc::KillActorRequest &request,
     send_reply_callback(Status::Invalid(msg), nullptr, nullptr);
     return;
   }
-  RAY_LOG(INFO) << "Got KillActor, shutting down...";
-  Shutdown();
+  RAY_LOG(INFO) << "Got KillActor, exiting immediately...";
+  if (log_dir_ != "") {
```

Comment:
why need this if statement?

Reply:
A bit ugly, but `glog` is only initialized if `log_dir_ != ""` (same check as in the destructor)